File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n=nums.length;
        int[] ls=new int[n];
        int[] rs=new int[n];
        int[] ans=new int[n];
        ls[0] = 1;
        rs[n-1]=1;
        for(int i=0;i<n-1;i++){
            ls[i+1]=ls[i]*nums[i];

        }
        for(int i=n-2; i>= 0; i--){
            rs[i]=rs[i+1]*nums[i+1];
        }
        for(int i=0;i< n;i++){
            ans[i]=ls[i]*rs[i];
        }
        return ans;
    }
}  

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefix=0
        count={0:1}
        res=0

        for i in nums:
            prefix+=i
            if prefix-k in count:
                res+=count[prefix-k]
            count[prefix]=count.get(prefix,0)+1

        return res
